### ISSUE TITLE
YAM doesn't check signature on files created with YAM

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#4]
**Originally by _amigasurfer@users.sourceforge.net_ on 2010-03-03 14:54:20 +0100**

___
I wrote an e-mail with YAM, signed it with PGP and send it to a mailinglist.

Then I tried to check the signatur. If I open this message or choose in the menubar to check the signatur I will get no message. I thought no error message says it would be good. But then I modified a character with a hex-editor. Same result, no error message about the now invalid signatur.

I have the impression, that YAM save header, text and signatur after converting quoted printable at T: in three separate files, but then only send the file with the signatur to PGP, not the file with the signed text. But I can't verify this with SnoopDos, because SnoopDos does only catch 99 characters of the command send from YAM to PGP.

Attached you will find this e-mail for testing.

Same problem appears e.g. with e-mails generated with User-Agent: Mutt/1.5.11, e.g. Message-ID: <20070311171652.GA8118@beverly.kleinbus.org> at the NetBSD-Amiga Mailinglist port-amiga@netbsd.org.

**Comment by jens-maus** [comment]
**Originally by _amigasurfer_ on 2010-03-03 14:55:47 +0100**

___
**Attachment added:** [`OIwB0gAKbV4=.001,R`](https://yam.ch/attachments/4/OIwB0gAKbV4=.001,R) (6.9 KiB)

**Comment by tboeckel** [comment]
**Originally on 2010-03-03 14:56:56 +0100**

___
To be sure which file PGP is fed with just create a "util" log and post it
here. See
http://faq.yam.ch/content/12/54/en/basics-about-yam_s-debugging-facilities.html
for more information about YAM's debugging facility. The 2.6 nightly builds
will log some information, but even with 2.5 you will see the used file
names.

**Comment by tboeckel** [comment]
**Originally on 2010-03-03 14:57:40 +0100**

___
**Attachment added:** [`YAM.debug.zip`](https://yam.ch/attachments/4/YAM.debug.zip) (101.4 KiB)

**Comment by jens-maus** [comment]
**Originally by _amigasurfer_ on 2010-03-03 14:58:53 +0100**

___
I made a "util" log with YAM 2.5 debug and the env yamdebug set to "util
@all".
Attached you will find the whole debug.log as zipped file.

The lines calling PGP are:


```
YAM_UT.c:5480:Entering PGPCommand
YAM_UT.c:5482:[pgpv] [T:YAMr00000014-p2.txt -o T:YAMt00000015.tmp
+batchmode=1 +force +language=us] - flags: 48
YAM_UT.c:5603:Entering Busy
```

**Comment by tboeckel** [comment]
**Originally on 2012-10-26 22:29:39 +0200**

___
This should finally be fixed by the next nightly build (see r6350 to r6359).

**Comment by jens-maus** [comment]
**Originally on 2012-10-28 19:28:41 +0100**

___
reopening this ticket as recent tests show that now with every execution of the PGP command output is shown in a console window. Thus, the output is not supressed anymore.

**Comment by tboeckel** [comment]
**Originally on 2012-10-28 20:46:41 +0100**

___
In (767215d):
 * YAM_UT.c: reworked the command output redirection to redirect the output of PGP to NIL: as it seems to use stderr for the general output. This closes #4 again.
